src/software/ai/hl/stp/strategy/strategy.cpp

```diff
@@ -0,0 +1,147 @@
+#include "software/ai/hl/stp/strategy/strategy.h"
+#include "software/ai/passing/eighteen_zone_pitch_division.h"
+#include "software/ai/passing/pass_generator.hpp"
+#include "software/ai/passing/pass_with_rating.h"
+#include "software/time/time.h"
+
+Strategy::Strategy(const TbotsProto::AiConfig& ai_config)
+    : end_analysis_(false)
+{
+    updateAiConfig(ai_config);
+
+    passing_thread_ = std::thread(&Strategy::evaluatePassOptions, this);
+}
+
+Strategy::~Strategy()
+{
+    end_analysis_ = true;
+
+    passing_thread_.join();
+}
+
+void Strategy::evaluatePassOptions()
+{
+    {
+        std::unique_lock<std::mutex> lock(world_lock_);
+        world_available_cv_.wait(lock, [&] { return current_world_.has_value(); });
+    }
+
+    while (!end_analysis_)
+    {
+        std::scoped_lock lock(pass_generator_lock_, world_lock_);
+        latest_pass_eval_ = std::make_shared<PassEvaluation<EighteenZoneId>>(
+                pass_generator_->generatePassEvaluation(current_world_.value()));
+        pass_available_cv_.notify_one();
```

Comment:
We should update the locked regions to cover as little as possible. e.g. This seems to help performance
```suggestion
        std::optional<World> passing_world;
        {
            std::scoped_lock lock(world_lock_);
            passing_world = current_world_.value();
        }
        auto pass = std::make_shared<PassEvaluation<EighteenZoneId>>(
                pass_generator_->generatePassEvaluation(passing_world.value()));
        {
            std::scoped_lock lock(pass_generator_lock_);
            latest_pass_eval_ = pass;
        }
        pass_available_cv_.notify_one();
```